src/Clerk.ts

```diff
@@ -204,36 +223,63 @@ export default class Clerk {
 
       try {
         const sessionToken = cookies.get('__session');
-
         Logger.debug(`sessionToken: ${sessionToken}`);
 
-        if (!sessionToken) {
-          throw new HttpError(401, 'No session cookie found', undefined)
+        let headerToken;
+        if (req.headers) {
+          headerToken = (req.headers['Authorization'] || req.headers['authorization']) as string
         }
 
-        let sessionId: any = req.query._clerk_session_id;
+        if (isAuthV2Request(this, headerToken) || isAuthV2Request(this, sessionToken)) {
+          // Set Clerk session claims on request
+          // TBD Set on state / locals instead?
+
+          try {
+            // @ts-ignore
+            req.sessionClaims = await this.verifyToken(headerToken);
```

Comment:
Maybe rename this to `currentSID`? How do we do this across middlewares? The name of the key on the request object should be the same across ruby, python, go.

Reply:
I came up with this name but i don't think is the best. Open to any naming suggestion. Note that this should be different that the already session field, as at least in Go sdk we can't assign a different type in the same context key without being an `interface{}`, which will confuse the users

Reply:
Let's go with currentSID then for now. I don't have a better suggestion. @colinclerk Any ideas?